### PR TITLE
Add Input constraint labels

### PR DESCRIPTION
**Pull request by sdasgup3**
Added Input constraint labels to help refer to input constraint while
1. writing PR description like [this]( https://github.com/openxla/stablehlo/pull/996#discussion_r1096336612) which makes sure all the constraints  have corresponding verification logic and tests.
2. Adding comments in tests and implementation of verifiers.

### CODE REVIEW

File: docs/spec.md [hunk 1]
```diff
@@ -468,9 +468,9 @@ tensor. Depending on the element type, does the following:
 
 #### Inputs
 
-| Name      | Type                                                      | Constraints |
-|-----------|-----------------------------------------------------------|-------------|
-| `operand` | tensor of signed integer, floating-point, or complex type | (C1), (C2)  |
+| Label | Name      | Type                                                      | Constraints |
+|-------|-----------|-----------------------------------------------------------|-------------|
+| (I0)  | `operand` | tensor of signed integer, floating-point, or complex type | (C1), (C2)  |
```
Comment (burmako) [inline docs/spec.md:473]: Let's start numbering from 1, i.e. `(I1)` etc, to be consistent with constraint labels?

**Review by burmako — CHANGES_REQUESTED** [review]

File: docs/spec.md (hunk 1 above)
Comment (sdasgup3) [reply docs/spec.md]: done.

File: docs/spec.md [hunk 2]
```diff
@@ -468,9 +468,9 @@ tensor. Depending on the element type, does the following:
 
 #### Inputs
 
-| Name      | Type                                                      | Constraints |
-|-----------|-----------------------------------------------------------|-------------|
-| `operand` | tensor of signed integer, floating-point, or complex type | (C1), (C2)  |
+| Label | Name      | Type                                                      | Constraints |
+|---|-----------|-----------------------------------------------------------|-------------|
+ | (I1) | `operand` | tensor of signed integer, floating-point, or complex type | (C1), (C2)  |
```
Comment (burmako) [inline docs/spec.md:473]: The tables look a bit wonky now: 1) the second row should be extended from 3 hyphens to 6 hyphens in the first column, 2) subsequent rows shouldn't start with a whitespace.

**Review by burmako — APPROVED** [review]
LGTM except for the formatting comment below.

File: docs/spec.md (hunk 2 above)
Comment (sdasgup3) [reply docs/spec.md]: Ah, my bad. Fixed.

**Review by ghpvnist — APPROVED** [review]

**Pull request merged**